### PR TITLE
Handle Nullchk operator

### LINKED ISSUE #366: java.lang.RuntimeException: whoops, better handle OTHER null
Encounter error when use Gradle to compile google android code with Error Prone plugin:
```
src/com/android/providers/settings/SettingsProvider.java:1487: error: An unhandled exception was thrown by the Error Prone static analysis plugin.
return settingsState.new Setting(ssaidSetting) {
^

error-prone version: 2.3.3
BugPattern: NullAway

 Stack Trace:
 java.lang.RuntimeException: whoops, better handle OTHER null
at com.uber.nullaway.NullAway.mayBeNullExpr(NullAway.java:1952)
at com.uber.nullaway.NullAway.handleInvocation(NullAway.java:1352)
at com.uber.nullaway.NullAway.matchNewClass(NullAway.java:354)
at com.google.errorprone.scanner.ErrorProneScanner.processMatchers(ErrorProneScanner.java:433)
at com.google.errorprone.scanner.ErrorProneScanner.visitNewClass(ErrorProneScanner.java:759)
at com.google.errorprone.scanner.ErrorProneScanner.visitNewClass(ErrorProneScanner.java:150)
at com.sun.tools.javac.tree.JCTree$JCNewClass.accept(JCTree.java:1705)
```

Gradle compilation passed without error-prone plugin, but failed if plugin integrated.

Relevant code fragment:

```
private Setting mascaradeSsaidSetting(SettingsState settingsState, Setting ssaidSetting) {
    if (ssaidSetting != null) {
        return settingsState.new Setting(ssaidSetting) {
              @override
               public int getKey() {
                    final int userId = getUserIdFromKey(super.getKey());
                    return makeKey(SETTINGS_TYPE_SECURE, userId);
                  }

             @Override
             public String getName() {
                 return Settings.Secure.ANDROID_ID;
              }
          };
      }
    return null;
}
```

### PR DESCRIPTION
**Pull request by shas19**
Handles `Nullchk` unary operator. 

Fixes #366

### CODE REVIEW

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java [hunk 1]
```diff
@@ -2103,6 +2104,11 @@ private static ExpressionTree stripParensAndCasts(ExpressionTree expr) {
         expr = ((TypeCastTree) expr).getExpression();
         someChange = true;
       }
+
+      // Strips Nullchk operator
+      if (expr.getKind().equals(OTHER) && expr instanceof JCTree.JCUnary) {
+        expr = ((JCTree.JCUnary) expr).getExpression();
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/NullAway.java]:
```suggestion
        expr = ((JCTree.JCUnary) expr).getExpression();
        someChange=true;
```
Need to set the `someChange` variable

File: nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java [hunk 2]
```diff
@@ -378,7 +378,10 @@ private ClassTree findEnclosingLocalOrAnonymousClass(ClassTree classTree) {
   @Override
   public TransferResult<Nullness, NullnessStore> visitNullChk(
       NullChkNode nullChkNode, TransferInput<Nullness, NullnessStore> input) {
-    throw new RuntimeException("we should never see this");
+    SubNodeValues values = values(input);
+    Nullness nullness =
+        hasPrimitiveType(nullChkNode) ? NONNULL : values.valueOfSubNode(nullChkNode.getOperand());
```
Comment (msridhar) [inline nullaway/src/main/java/com/uber/nullaway/dataflow/AccessPathNullnessPropagation.java]: I don't think the compiler will ever insert a `NullChk` around a node of primitive type, but we can leave the check out of paranoia 😄

**Comment by msridhar** [comment]
Also looks like tests are failing on JDK 11.  @shas19 are you set up to test that?

File: nullaway/src/main/java/com/uber/nullaway/NullAway.java (hunk 1 above)
Comment (lazaroclapp) [reply nullaway/src/main/java/com/uber/nullaway/NullAway.java]: Modulo adding this patch, this PR looks good to me :)

**Comment by shas19** [comment]
```
  class AnotherClass{
    class Inner{
      int x;
    }
  }

  AnotherClass.Inner someMethod(){
    AnotherClass anotherClass = new AnotherClass();
    return anotherClass.new Inner(){
      int getX(){ return x;}
    };
  }
```

On this test there is behavior difference between JDK8 and JDK11. When `matchNewClass` is caleed, on JDK8 the value of `mthodSymbol` is `Inner(AnotherClass)` and on JDK11 it is ` (AnotherClass)`.  In case of JDK8, both actualParams and formalParams are empty lists. While in case of JDK11 formalParams has one value `x_0`. The code in `handleInvocation` is written with the assumption that `actualParams` and `formalParams` are of same size. So I have added a check to stop checking if that is not handled. 

But yeah, we will need to investigate further, what is the exact difference in the behavior of JDK8 and JDK11. If you think we should fix this after we know the reason clearly then we can hold this one.

**Comment by msridhar** [comment]
@shas19 can we somehow narrow the bailout check to the very specific case this is occurring (invoking a non-static inner class constructor)?  Ideally we'd also only do the check on JDK 11+.

**Comment by msridhar** [comment]
@shas19 do you have cycles to finish this one up?

**Pull request merged**